Core/include/Acts/Seeding/PathSeeder.hpp

```diff
@@ -0,0 +1,185 @@
+    m_cfg.sourceLinkGrid->initialize(gctx, sourceLinks);
+
+    // Get plane of the telescope
+    // sensitive surfaces
+    int bin0 = static_cast<int>(BinningValue::binX);
+    int bin1 = static_cast<int>(BinningValue::binY);
+    if (m_cfg.orientation == BinningValue::binX) {
+      bin0 = static_cast<int>(BinningValue::binY);
+      bin1 = static_cast<int>(BinningValue::binZ);
+    } else if (m_cfg.orientation == BinningValue::binY) {
+      bin0 = static_cast<int>(BinningValue::binX);
+      bin1 = static_cast<int>(BinningValue::binZ);
+    }
+
+    // Create the seeds
+    std::vector<Seed> seeds;
+    for (const auto& sl : sourceLinks) {
+      Vector3 globalPos = m_cfg.sourceLinkCalibrator(gctx, sl);
+
+      // Check if the hit is in the
+      // first tracking layer
+      if (!m_cfg.firstLayerExtent.contains(globalPos)) {
+        continue;
+      }
+
+      // Get the IP parameters
+      auto [ipP, ipVertex, ipDir, flDir] =
+          m_cfg.trackEstimator(gctx, globalPos);
+
+      // Intersect with the surfaces
+      std::vector<std::pair<GeometryIdentifier, Vector3>> intersections =
+          m_cfg.intersectionFinder(gctx, globalPos, flDir, ipP);
+
+      // Continue if no intersections
+      if (intersections.empty()) {
+        continue;
+      }
+      // Vector to store the source links
+      std::vector<SourceLink> seedSourceLinks;
+
+      // Store the pivot source link
+      seedSourceLinks.push_back(sl);
+
+      // Iterate over the intersections
+      // and get the source links
+      // in the subsequent layers
+      for (auto& [geoId, refPoint] : intersections) {
+        // Get the path width
+        auto [pathWidth0, pathWidth1] = m_cfg.pathWidthProvider(gctx, geoId);
+
+        // Get the bounds of the path
+        ActsScalar top0 = refPoint[bin0] + pathWidth0;
+        ActsScalar bot0 = refPoint[bin0] - pathWidth0;
+        ActsScalar top1 = refPoint[bin1] + pathWidth1;
+        ActsScalar bot1 = refPoint[bin1] - pathWidth1;
+
+        // Get the lookup table for the source links
+        auto grid = m_cfg.sourceLinkGrid->getSourceLinkTable(geoId);
+
+        // Get the range of bins to search for source links
+        auto botLeftBin = grid.localBinsFromPosition(Vector2(bot0, bot1));
+        auto topRightBin = grid.localBinsFromPosition(Vector2(top0, top1));
+
+        // Get the source links from the lookup table
+        // by iterating over the bin ranges
+        auto currentBin = botLeftBin;
+        while (currentBin.at(1) <= topRightBin.at(1)) {
+          while (currentBin.at(0) <= topRightBin.at(0)) {
+            auto sourceLinksToAdd = grid.atLocalBins(currentBin);
+
+            seedSourceLinks.insert(seedSourceLinks.end(),
+                                   sourceLinksToAdd.begin(),
+                                   sourceLinksToAdd.end());
+            currentBin.at(0)++;
+          }
+          currentBin.at(1)++;
+          currentBin.at(0) = botLeftBin.at(0);
+        }
+      }
```

Comment:
I wonder if we should equip the grid with a range lookup?

Reply:
I will discuss this with @paulgessinger.

Reply:
I would suggest to merge it as is for practicality. I would create a separate PR for the range search, and when it's ready just modify the relevant parts of the codebase